converters/template/cmd/main.go

```diff
@@ -20,15 +20,38 @@
 package main
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
 	sourceapis "github.com/crossplane-contrib/provider-aws/apis"
-	"github.com/upbound/extensions-migration/converters/common"
-	provideraws "github.com/upbound/extensions-migration/converters/provider-aws"
 	targetapis "github.com/upbound/provider-aws/apis"
 	"github.com/upbound/upjet/pkg/migration"
+	"gopkg.in/alecthomas/kingpin.v2"
 	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+
+	"github.com/upbound/extensions-migration/converters/common"
+	provideraws "github.com/upbound/extensions-migration/converters/provider-aws"
 )
 
 func main() {
+	// Common CLI Flags
+	// They can be extended according to requirements
+	var (
+		app            = kingpin.New(filepath.Base(os.Args[0]), "Upbound migration plan generator for migrating Kubernetes objects from community providers to official providers.").DefaultEnvars()
+		planPath       = app.Flag("plan-path", "Path where the generated migration plan will be stored").Short('p').Default("migration_plan.yaml").String()
+		sourcePath     = app.Flag("source-path", "Path of the root directory for the filesystem source").Short('s').String()
+		kubeconfigPath = app.Flag("kubeconfig", "Path of the kubernetes config file. Defaults to ~/.kube/config ").String()
+	)
+	if len(*kubeconfigPath) == 0 {
```

Comment:
same: do we need to do nil checks before dereferencing or do they have default values?

Reply:
See above answer.